### PR TITLE
Breakpoint validation for Microsoft issue #75

### PR DESCRIPTION
**Pull request by erwindevreugd**
Added validation for positioning breakpoints. Breakpoints are no longer settable on empty lines or comments. Also wraps the code in a breakpoint highlight.
Issue: https://github.com/Microsoft/poshtools/issues/75

### CODE REVIEW

**Comment by adamdriscoll** [comment]
So far it looks good but I want to pull it and play a little bit. One comment is that it would be good to wrap some tests around it since there is a lot of if\else type logic that could be easy to mess up if it were to be worked on in the future.

**Comment by erwindevreugd** [comment]
At the moment I'm having some trouble getting the existing unit test to work on my machine (using NCruch) as soon as I can figure out what the issue is I'll add some unit tests.

If your going to play with this please pay attention to which lines you are allowed to set breakpoints and where the span shows up. I've did my best to match them to the way VS does its debugvalidation and positioning on C# code and to where the actual debugger hits, but there may still be some discrepancies as there are a lot of options. Changes in current behavior are easily made.

**Comment by erwindevreugd** [comment]
An other option would be to chose to only show the breakpoint in the magin and not as a span around the code. That would eliminate a lot of if statements.

**Comment by adamdriscoll** [comment]
That's a good point. The ISE looks like it doesn't even validate on comments and highlights the whole line. I could go either way. I do like the idea of highlighting exactly what is executable code. 

image

**Comment by erwindevreugd** [comment]
And i like the idea of being on par with ISE but it is VS after al.
Let me know what you think and find and I can continue work on this before merging.

**Comment by adamdriscoll** [comment]
Will do. Totally agree with the fact this is VS and not ISE but we've been trying to check what the ISE is doing when we can. In this case, I like how VS does it better.

**Comment by adamdriscoll** [comment]
Looks good to me! Seems to highlight all the correct spans etc. @AndreSayreMSFT @HuanhuanSunMSFT have any comments?

File: PowerShellTools/LanguageService/PowerShellLanguageInfo.cs [hunk 1]
```diff
@@ -149,26 +154,15 @@ public int ResolveName(string pszName, uint dwFlags, out IVsEnumDebugName ppName
...
-            return VSConstants.E_NOTIMPL;
+            var dte2 = (DTE2)Package.GetGlobalService(typeof(SDTE));
+            if (dte2.ActiveDocument == null || !LanguageUtilities.IsPowerShellExecutableScriptFile(dte2.ActiveDocument.FullName))
+            {
+                return VSConstants.S_FALSE;
+            }
+
+            var breakpointValidationService = new BreakpointValidationService(_editorAdaptersFactoryService);
+
+            return breakpointValidationService.ValidateBreakpointLocation(pBuffer, iLine, ref pCodeSpan);
```
Comment (HuanhuanSunMSFT) [inline PowerShellTools/LanguageService/PowerShellLanguageInfo.cs]: Is it possible that pCodeSpan is null or empty which may cause exception?

**Comment by HuanhuanSunMSFT** [comment]
Looks good to me.

File: PowerShellTools/LanguageService/PowerShellLanguageInfo.cs (hunk 1 above)
Comment (erwindevreugd) [inline PowerShellTools/LanguageService/PowerShellLanguageInfo.cs]: The MSDN documentation isn't exactly clear on that. Their sample does do null checking but the original code didn't and based on that I don't think it's ever null.

**Pull request merged**